### ISSUE TITLE
Errors when building the project

### ISSUE THREAD

**Issue by noncom** [jakubtomsu/dungeon-of-quake#2]
Hi! I tried to build the project, but following the instruction I'm getting:

```powershell
PS C:\work\exp\odin\test-ws\dungeon-of-quake> build.bat run
[vcvarsall.bat] Environment initialized for: 'x64'
main.cpp
libtommath.cpp
c1xx: fatal error C1083: Cannot open source file: 'src\main.cpp': No such file or directory
c1xx: fatal error C1083: Cannot open source file: 'src\libtommath.cpp': No such file or directory
```

Is there something I should add to the project before building it?

I'm not very familiar with the C/C++ ecosystem, but I found that `libtonmath` is a separate math library.
Not sure what the `main.cpp` would be related to though...

**Comment by jakubtomsu** [comment]
It looks like you are somehow calling the `build.bat` file from Odin compiler...? Our build script doesn't call vcvarsall.bat etc and doesn't try to build any cpp files.

You can try building with `odin build doq -out:build/dungeon_of_quake.exe` or something like that. But tbh I haven't touched this project in a really long time, it's possible it doesn't compile anymore. I'll try to take a look at it soon-ish, and ideally update it to newest Odin version.

**Comment by noncom** [comment]
Hey, that was my mistake. I rarely work with PowerShell, and I forgot that it needs to be `./build.bat`, and not just `build.bat`.
So I have no idea what it was actually doing before...
But also, `./build.bat` gives an error, though a much more sensible one, probably because of changes to what arguments Odin expects.

By using the build command you provided I was able to actually launch the build process.

But still, true to the original title of this issue, the build resulted in code errors.
I tried fixing them best I could, and the game seems to be able to build and start now, but it's not playable because turning doesn't work, and the player seems to be constantly bouncing off the surface.

I haven't yet investigated why that might be the case. The project is already by itself a lot of material for me to learn.

I have created a MR here: https://github.com/jakubtomsu/dungeon-of-quake/pull/3 but it's more just for reference so you don't waste time on searching for where the hell did the stuff from Raylib go. I'm not sure that the changes that I made are idiomatic or even correct since I'm new to Odin, and I have very little experience with C and Raylib.

Also, thank you very much for creating the project and sharing it!
As the amount of examples for using Odin is so limited, and this one covers so much, it is a great help and inspiration.
It is awesome if it is updated to keep it being most relevant and helpful 😄

**Comment by jakubtomsu** [comment]
Thanks for the PR! I'll take a look at it and try to incorporate it into the project. I should have some free time over the weekend to take a look at it.

FYI, I did this project like 2 years ago. It was the first non trivial (but still fairly simple) piece of software I wrote in Odin. The code is actually pretty bad, and it's not an example of "idiomatic" Odin. I still think it can be a useful resource for learning Odin. Just keep in mind some things might be a bit scuffed.

The primary reason I haven't updated the project in a long time is that whenever I've thought about the things I should upgrade, the scope got pretty big. So I haven't really found the time to do a big update yet.
